### ISSUE TITLE
`createFactory ` is deprecated Warning in react 16.13.1

### ISSUE THREAD

**Issue by macrozone** [marmelab/react-admin#4731]
**What you were expecting:**

No warnings

**What happened instead:**

```
      Warning: React.createFactory() is deprecated and will be removed in a future ma
jor release. Consider using JSX or use React.createElement() directly instead.

      at printWarning (packages/dataprovider/node_modules/react/cjs/react.development
.js:315:30)
      at warn (node_modules/react/cjs/react.development.js:278:5)
      at createFactoryWithValidation (node_modules/react/cjs/react.development.js:1827:7)
      at node_modules/recompose/shouldUpdate.js:25:44
      at Object.pure [as default] (node_modules/recompose/pure.js:29:97)
      at Object.<anonymous> (node_modules/ra-core/lib/util/FieldTitle.js:19:33)
```

**Steps to reproduce:**

update to react 16.13.1

**Related code:**

its actually an upstream issue from recompose: https://github.com/acdlite/recompose/pull/795

unforunatly, the pull request is not merged and it does not seem to be active.

an alternative would be to use React.memo instead of pure from recompose 

**Environment**

* React-admin version: 3.4.1
* Last version that did not exhibit the issue (if applicable):
* React version: 16.13.1
* Browser: chrome 80
* Stack trace (in case of a JS error): see above

**Comment by djhi** [comment]
Yes, I noticed those annoying warnings as well. Now that most of our components use hooks, we can probably remove recompose.

**Comment by fzaninotto** [comment]
Fixed by #4786
